### PR TITLE
WHO 9 Time Travel Cards

### PR DESCRIPTION
**Pull request by Simisays**
This should be all the time travel cards.

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/the_parting_of_the_ways.txt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+Name:The Parting of the Ways
+ManaCost:4 R R
+Types:Enchantment Saga
+K:Saga:3:DBDig,DBTimeTravel,DBDestroy
+SVar:DBDig:DB$ Dig | DigNum$ 5 | ChangeNum$ All | ChangeValid$ Card | DestinationZone$ Exile | RememberChanged$ True | SubAbility$ DBRepeat | SpellDescription$ Exile the top five cards of your library. For each nonland card exiled this way, put a number of time counters on that card equal to its mana value. If it doesn't have suspend, it gains suspend.
+SVar:DBRepeat:DB$ RepeatEach | UseImprinted$ True | RepeatCards$ Card.IsRemembered+nonLand | Zone$ Exile | RepeatSubAbility$ DBPutCounter | SubAbility$ DBCleanup
+SVar:DBPutCounter:DB$ PutCounter | Defined$ Imprinted | CounterType$ TIME | CounterNum$ X | SubAbility$ GiveSuspend
+SVar:GiveSuspend:DB$ Pump | Defined$ Remembered | KW$ Suspend | PumpZone$ Exile | Duration$ Permanent | SubAbility$ DBCleanup
+SVar:DBCleanup:DB$ Cleanup | ClearImprinted$ True | ClearRemembered$ True
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_parting_of_the_ways.txt:9]: don't need to cleanup imprinted when only RepeatEach used it

File: forge-gui/res/cardsfolder/upcoming/the_parting_of_the_ways.txt [hunk 2]
```diff
@@ -0,0 +1,14 @@
+Name:The Parting of the Ways
+ManaCost:4 R R
+Types:Enchantment Saga
+K:Saga:3:DBDig,DBTimeTravel,DBDestroy
+SVar:DBDig:DB$ Dig | DigNum$ 5 | ChangeNum$ All | ChangeValid$ Card | DestinationZone$ Exile | RememberChanged$ True | SubAbility$ DBRepeat | SpellDescription$ Exile the top five cards of your library. For each nonland card exiled this way, put a number of time counters on that card equal to its mana value. If it doesn't have suspend, it gains suspend.
+SVar:DBRepeat:DB$ RepeatEach | UseImprinted$ True | RepeatCards$ Card.IsRemembered+nonLand | Zone$ Exile | RepeatSubAbility$ DBPutCounter | SubAbility$ DBCleanup
+SVar:DBPutCounter:DB$ PutCounter | Defined$ Imprinted | CounterType$ TIME | CounterNum$ X | SubAbility$ GiveSuspend
+SVar:GiveSuspend:DB$ Pump | Defined$ Remembered | KW$ Suspend | PumpZone$ Exile | Duration$ Permanent | SubAbility$ DBCleanup
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_parting_of_the_ways.txt:8]: wrong/incomplete Defined

File: forge-gui/res/cardsfolder/upcoming/the_tenth_doctor.txt [hunk 3]
```diff
@@ -0,0 +1,12 @@
+Name:The Tenth Doctor
+ManaCost:3 U R
+Types:Legendary Creature Time Lord Doctor
+PT:3/5
+T:Mode$ AttackersDeclared | AttackingPlayer$ You | Execute$ TrigExile | TriggerZones$ Battlefield | TriggerDescription$ Allons-y! — Whenever you attack, exile cards from the top of your library until you exile a nonland card. Put three time counters on it. If it doesn't have suspend, it gains suspend.
+SVar:TrigExile:DB$ DigUntil | Valid$ Card.nonLand | ValidDescription$ nonland | FoundDestination$ Exile | RevealedDestination$ Exile | RememberFound$ True | SubAbility$ DBPutCounter
+SVar:DBPutCounter:DB$ PutCounter | Defined$ Remembered | CounterType$ TIME | CounterNum$ 3 | SubAbility$ GiveSuspend
+SVar:GiveSuspend:DB$ Pump | Defined$ Remembered | KW$ Suspend | PumpZone$ Exile | Duration$ Permanent | SubAbility$ DBCleanup
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_tenth_doctor.txt:8]: same problem

File: forge-gui/res/cardsfolder/upcoming/the_wedding_of_river_song.txt [hunk 4]
```diff
@@ -0,0 +1,11 @@
+Name:The Wedding of River Song
+ManaCost:2 W
+Types:Sorcery
+A:SP$ Draw | DrawNum$ 2 | SubAbility$ DBExile | SpellDescription$ Draw two cards, then you may exile a nonland card from your hand with a number of time counters on it equal to its mana value. Then target opponent does the same. Cards exiled this way that don't have suspend gain suspend.
+SVar:DBExile:DB$ ChangeZone | Origin$ Hand | Destination$ Exile | ChangeType$ Card.nonLand | WithCountersType$ TIME | WithCountersAmount$ X | RememberChanged$ True | SubAbility$ OppExile
+SVar:OppExile:DB$ ChangeZone | Origin$ Hand | Destination$ Exile | ValidTgts$ Opponent | Chooser$ Targeted | Hidden$ True | Mandatory$ True | WithCountersType$ TIME | WithCountersAmount$ Y | ChangeType$ Card.nonLand | Imprint$ True |  SubAbility$ GiveSuspend
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_wedding_of_river_song.txt:6]: if first isn't Mandatory why would this be?

File: forge-gui/res/cardsfolder/upcoming/the_tenth_doctor.txt (hunk 3 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/the_tenth_doctor.txt]: you could have just written `Defined$ Remembered.withoutSuspend`
(one of the smart ways to combine Defined + Valid syntax)

File: forge-gui/res/cardsfolder/upcoming/the_parting_of_the_ways.txt (hunk 2 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/the_parting_of_the_ways.txt]: still looks pretty awkward you're Pumping each loop when only the first will do anything now
I'd recommend adding `ClearRemembered` on the RepeatEach so you can save the extra effect too

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/the_parting_of_the_ways.txt]: Would it be better to have the pumpall outside the loop ?

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/the_parting_of_the_ways.txt]: well then you wouldn't be able to get rid of DBCleanup

File: forge-gui/res/cardsfolder/upcoming/the_tenth_doctor.txt (hunk 3 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/the_tenth_doctor.txt]: you're still using the wrong effect/param so you can't use the syntax

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/the_tenth_doctor.txt]: Ah yeah defined with pumpall is for players

File: forge-gui/res/cardsfolder/upcoming/the_parting_of_the_ways.txt (hunk 2 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/the_parting_of_the_ways.txt]: ok, just need to stop using Imprinted now and change the lines accordingly

File: forge-gui/res/cardsfolder/upcoming/all_of_history_all_at_once.txt [hunk 5]
```diff
@@ -0,0 +1,7 @@
+Name:All of History, All at Once
+ManaCost:2 U U
+Types:Sorcery
+A:SP$ Pump | Cost$ 2 U U | StackDescription$ None
+K:Time travel
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/all_of_history_all_at_once.txt:5]: what's with this KW?

File: forge-gui/res/cardsfolder/upcoming/the_wedding_of_river_song.txt [hunk 6]
```diff
@@ -0,0 +1,11 @@
+Name:The Wedding of River Song
+ManaCost:2 W
+Types:Sorcery
+A:SP$ Draw | DrawNum$ 2 | SubAbility$ DBExile | SpellDescription$ Draw two cards, then you may exile a nonland card from your hand with a number of time counters on it equal to its mana value. Then target opponent does the same. Cards exiled this way that don't have suspend gain suspend.
+SVar:DBExile:DB$ ChangeZone | Origin$ Hand | Destination$ Exile | ChangeType$ Card.nonLand | WithCountersType$ TIME | WithCountersAmount$ X | RememberChanged$ True | SubAbility$ OppExile
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_wedding_of_river_song.txt:5]: pretty sure it's templated so opponent also gets the first part (draw)

File: forge-gui/res/cardsfolder/upcoming/the_wedding_of_river_song.txt [hunk 7]
```diff
@@ -0,0 +1,11 @@
+Name:The Wedding of River Song
+ManaCost:2 W
+Types:Sorcery
+A:SP$ Draw | DrawNum$ 2 | SubAbility$ DBExile | SpellDescription$ Draw two cards, then you may exile a nonland card from your hand with a number of time counters on it equal to its mana value. Then target opponent does the same. Cards exiled this way that don't have suspend gain suspend.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_wedding_of_river_song.txt:4]: Reviewing is really hard when you make up params like DrawNum$ 😢

File: forge-gui/res/tokenscripts/w_1_1_human_noble_vanishing_prevention.txt [hunk 8]
```diff
@@ -0,0 +1,8 @@
+Name:Human Noble
```
Comment (tool4ever) [inline forge-gui/res/tokenscripts/w_1_1_human_noble_vanishing_prevention.txt:1]: wrong names

Comment (Simisays) [reply forge-gui/res/tokenscripts/w_1_1_human_noble_vanishing_prevention.txt:1]: Do all tokens need the _token_ at the end ? 

image

Comment (tool4ever) [reply forge-gui/res/tokenscripts/w_1_1_human_noble_vanishing_prevention.txt:1]: > 111.4. A spell or ability that creates a token sets both its name and its subtype(s). If the spell or ability doesn't specify the name of the token, its name is the same as its subtype(s) plus the word "Token."

**Review by tool4ever — APPROVED** [review]

**Pull request merged**